src/App.tsx

```diff
@@ -1,156 +1,137 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import classNames from 'classnames';
+import React, { useEffect, useMemo, useState } from 'react';
+import * as todosService from './api/todos';
 
-const USER_ID = 0;
+import { TodoHeader } from './components/TodoHeader';
+import { TodoBody } from './components/TodoBody';
+import { TodoFooter } from './components/TodoFooter';
+import { FilterBy } from './types/FilterBy';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [newTodoId, setNewTodoId] = useState<number[]>([]);
+
+  // Request todos from server on first render
+  useEffect(() => {
+    todosService.getTodos(todosService.USER_ID)
+      .then(setTodos)
+      .catch((error) => {
+        setErrorMessage('load');
+        throw error;
+      })
+      .finally(() => {
+        setErrorMessage('');
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  // Filtering todos by FilteringBy without request on server
+  const filteringBy = useMemo(() => {
+    switch (filterBy) {
+      case FilterBy.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterBy.COMPLETED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [filterBy, todos]);
+
+  // Update or Edit todo on server
+  const updateTodo = async (todoToUpdate: Todo) => {
+    setNewTodoId(currentIDs => [...currentIDs, todoToUpdate.id]);
+
+    try {
+      const updatedTodo = await todosService
+        .updateTodo(todoToUpdate.id, todoToUpdate) as Todo;
+
+      const findIndexTodo = [...todos]
```

Comment:
```suggestion
      const findIndexTodo = [...todos]
```
You can remove this additional logic and do everything in setTodos